cluster/src/main/scala/io/snappydata/impl/LeadImpl.scala

```diff
@@ -234,6 +235,11 @@ class LeadImpl extends ServerImpl with Lead
         markLauncherRunning(if (startupString ne null) s"Starting $startupString" else null)
       }
 
+      // Add a URL classloader to the main thread so that new URIs can be added
+      val parent = Thread.currentThread().getContextClassLoader
+      urlclassloader = new ExtendibleURLClassLoader(parent)
+      Thread.currentThread().setContextClassLoader(urlclassloader)
```

Comment:
Do we have to set the classloader in start method?  I guess only query & job path should suffice?
You are already taking care of adding all urls at restart.